model/inline/src/main/scala/aqua/model/inline/raw/ApplyStreamMapRawInliner.scala

```diff
@@ -0,0 +1,366 @@
+      canonName <- mang(mapName, "canon")
+      result = VarModel(resultName, CanonStreamType(ScalarType.string))
+    } yield {
+      val getKeysTree = keysModel(
+        mapName = mapName,
+        mapType = mapType,
+        streamName = streamName,
+        iterName = iterName,
+        canonName = canonName,
+        result = result
+      )
+
+      val inline = Inline(predo = Chain.one(getKeysTree))
+
+      (result, inline)
+    }
+  }
+
+  private def getKeysStream[S: Mangler: Exports: Arrows: Config](
+    mapName: String,
+    mapType: StreamMapType
+  ): State[S, (VarModel, Inline)] = {
+    for {
+      streamName <- mang(mapName, "keys_stream_result")
+      iterName <- mang(mapName, "iter")
+      streamVar = VarModel(streamName, StreamType(ArrayType(mapType.element)))
+      // add resulted stream to restrict it after inlining
+      _ <- Exports[S].resolved(streamName, streamVar)
+    } yield {
+      val getKeysTree = getKeysStreamModel(
+        mapName = mapName,
+        mapType = mapType,
+        streamVar = streamVar,
+        iterName = iterName
+      )
+
+      val inline = Inline(predo = Chain.one(getKeysTree))
+
+      (streamVar, inline)
+    }
+  }
+
+  private def getStream[S: Mangler: Exports: Arrows: Config](
+    mapName: String,
+    mapType: StreamMapType,
+    idxVar: ValueModel
+  ): State[S, (VarModel, Inline)] = {
+    for {
+      streamName <- mang(mapName, "get_stream_result")
+      iterName <- mang(mapName, "iter")
+      value = VarModel(
+        streamName,
+        StreamType(mapType.element)
+      )
+      // add resulted stream to restrict it after inlining
+      _ <- Exports[S].resolved(streamName, value)
+    } yield {
+      val getStreamResultTree = getStreamModel(
+        mapName = mapName,
+        mapType = mapType,
+        idxVar = idxVar,
+        streamVar = value,
+        iterName = iterName
+      )
+
+      val inline = Inline(predo = Chain.one(getStreamResultTree))
+
+      (value, inline)
+    }
+  }
+
+  private def get[S: Mangler: Exports: Arrows: Config](
+    mapName: String,
+    mapType: StreamMapType,
+    idxVar: ValueModel
+  ): State[S, (VarModel, Inline)] = {
+    for {
+      resultName <- mang(mapName, "get_result")
+      canonMapName <- mang(mapName, "canon")
+      idxName <- mang(mapName, "idx")
+    } yield {
+      val getResultTree = getModel(
+        mapName,
+        mapType,
+        idxVar,
+        resultName,
+        canonMapName,
+        idxName
+      )
+
+      val inline = Inline(predo = Chain.one(getResultTree))
+      val value = VarModel(
+        resultName,
+        ArrayType(mapType.element)
+      )
+
+      (value, inline)
+    }
+  }
+
+  def apply[S: Mangler: Exports: Arrows: Config](
+    funcName: String,
+    mapName: String,
+    mapType: StreamMapType,
+    args: List[ValueModel]
+  ): State[S, (VarModel, Inline)] = {
+    StreamMapType
+      .funcByString(funcName)
+      .map(_ -> args)
```

Comment:
nit: with `import cats.syntax.functor.*`:
```suggestion
      .tupleRight(args)
```